### PR TITLE
Rules change: Update the plugin to Luke's rules

### PR DESCRIPTION
**Pull request by rrees**
Luke Hoyland has written some [new rules](https://docs.google.com/document/d/1IjTDH21WiyWSx63pjBDcbSn5WeRHM1X2oouuIMHVFdA/edit) which I have attempted to implement.

As this changes the output of the plugin it is going to be a major release.

### CODE REVIEW

**Comment by hmgibson23** [comment]
I think the rules might be linking to the wrong place

File: test/hyphen.spec.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+describe('Hyphen cleaining', () => {
+    givenContentOf('<p>content -- </p>', () => {
+        when('the command is executed', () => {
+            beforeEach(clickCleanup);
+
+            it('should replace the -- with an endash', () => {
+                return scribeNode.getInnerHTML().then((innerHTML) => {
+                    expect(innerHTML).to.include('<p>content – </p>');
```
Comment (hmgibson23) [inline test/hyphen.spec.js:35]: Is this an endash, looks like an emdash? But then they do look pretty similar.

Comment (hmgibson23) [inline test/hyphen.spec.js:35]: Sorry just looked at the rules - they do look very similar!

File: src/scribe-plugin-content-cleaner.js [hunk 2]
```diff
@@ -4,21 +4,15 @@ module.exports = function(scribe) {
...
-        },
-        (text) => {
-            return text.replace(/(<br\s*\/?>){3,}/gi, '<br>');
-        },
-        (text) => {
-            return text.replace(/<p>\s*?<br\s*\/?><\/p>/g, '');
-        }
+        (text) => text.replace(/(&nbsp;)/gi, " "),
+        (text) => text.replace(/\s+/g, " "),
+        (text) => text.replace(/ -- /g, " &ndash; "),
```
Comment (hmgibson23) [inline src/scribe-plugin-content-cleaner.js:9]: Are spaces around the dash definitely [correct](http://www.thepunctuationguide.com/en-dash.html)?

**Comment by hmgibson23** [comment]
Seems fine apart from my pedantic grammatical questions :+1:

**Comment by rrees** [comment]
I've update the rules link

File: src/scribe-plugin-content-cleaner.js (hunk 2 above)
Comment (rrees) [inline src/scribe-plugin-content-cleaner.js:9]: Guardian dashes seem to always have spaces around them in the style guide. Now whether the Guardian is _correct_ or not is beyond me.

File: test/hyphen.spec.js (hunk 1 above)
Comment (rrees) [inline test/hyphen.spec.js:35]: Ironically if I have pasted in the wrong character then all my tests might be for nothing!

**Pull request merged**